src/conn_sock.c

```diff
@@ -106,7 +106,10 @@ char *setup_console_socket(void)
 	if (listen(console_socket_fd, 128) < 0)
 		pexit("Failed to listen on console-socket");
 
-	return g_strdup(csname);
+	/* Move the ownership out of csname.  */
+	char *tmp = csname;
+	csname = NULL;
+	return tmp;
```

Comment:
couldn't we also just drop _cleanup_free_ when declaring csname?

Reply:
yes, good point.  I'll do that